homeassistant/components/slide_local/config_flow.py

```diff
@@ -0,0 +1,129 @@
+"""Config flow for slide_local integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from goslideapi.goslideapi import (
+    AuthenticationFailed,
+    ClientConnectionError,
+    ClientTimeoutError,
+    DigestAuthCalcError,
+    GoSlideLocal as SlideLocalApi,
+)
+import voluptuous as vol
+
+from homeassistant.components.zeroconf import ZeroconfServiceInfo
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_API_VERSION, CONF_HOST, CONF_MAC, CONF_PASSWORD
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.selector import (
+    BooleanSelector,
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from . import SlideConfigEntry
+from .const import CONF_INVERT_POSITION, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+BOOLEAN_SELECTOR = BooleanSelector()
+
+API_VERSION_SELECTOR = SelectSelector(
+    SelectSelectorConfig(
+        options=[
+            SelectOptionDict(value="1", label="API 1"),
+            SelectOptionDict(value="2", label="API 2"),
+        ],
+        mode=SelectSelectorMode.DROPDOWN,
+        translation_key=CONF_API_VERSION,
+    )
+)
+
+
+class SlideConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for slide_local."""
+
+    _entry: SlideConfigEntry | None = None
+    _mac: str = ""
+    _host: str = ""
+
+    VERSION = 1
+    MINOR_VERSION = 1
+
+    async def async_test_connection(
+        self, user_input: dict[str, str | int]
+    ) -> dict[str, str]:
+        """Reusable Auth Helper."""
+        slide = SlideLocalApi()
+        await slide.slide_add(
+            user_input[CONF_HOST],
+            user_input[CONF_PASSWORD],
+            user_input[CONF_API_VERSION],
+        )
+
+        try:
+            result = await slide.slide_info(user_input[CONF_HOST])
+            self._mac = result.get("mac", "")
```

Comment:
When can the Mac be empty?

Reply:
I'm not sure. Got this from the (CC)[https://github.com/ualex73/slide/blob/214c44be4f60ac071e1c44694bfd5b83ffeaa16d/custom_components/slide/cover.py#L279] and assumed, that it is a possibility.

Reply:
I would say fuck it we ball and just have it running for tests. If we notice that the mac indeed can be gone we should then discover what state the device is in and how we can fix that then